helix-term/src/commands.rs

```diff
@@ -3551,13 +3551,29 @@ fn open(cx: &mut Context, open: Open, comment_continuation: CommentContinuation)
 
     let mut ranges = SmallVec::with_capacity(selection.len());
 
-    let continue_comment_tokens =
-        if comment_continuation == CommentContinuation::Enabled && config.continue_comments {
+    let continue_comment_tokens = {
+        static SHEBANG_REGEX: Lazy<Regex> =
+            Lazy::new(|| Regex::new(&["^", SHEBANG].concat()).unwrap());
+
+        let curr_line_num = selection.primary().cursor_line(text);
+        let line = text.line(curr_line_num);
+
+        let is_not_shebang = curr_line_num > 0
+            || line
+                .as_str()
+                .map(|line| SHEBANG_REGEX.captures(line).is_some())
+                .unwrap_or(false);
+
+        if is_not_shebang
+            && comment_continuation == CommentContinuation::Enabled
+            && config.continue_comments
```

Comment:
I'm a bit unsure about this chunky boi is it fine if I move that into a private function within this module?